Fix unknown canonicalize_file_path on termux

Querela opened the pull request:
When trying to build on Termux, the `canonicalize_file_name` is somehow unknown and the build fails. Based on the [man pages](https://man7.org/linux/man-pages/man3/canonicalize_file_name.3.html), `realpath(path, NULL);` is equivalent, and compiles without issues. Text extraction seems to work fine.

Am I just missing some dependencies?

<details>
<summary>`make` error log</summary>

```bash
~/tools/epub2txt2 $ make
gcc -Wall -Wno-unused-result -O3  -DVERSION=\"2.10\" -DAPPNAME=\"epub2txt\" -MD -MF build/convertutf.deps -c -o build/convertutf.o src/convertutf.c            
gcc -Wall -Wno-unused-result -O3  -DVERSION=\"2.10\" -DAPPNAME=\"epub2txt\" -MD -MF build/epub2txt.deps -c -o build/epub2txt.o src/epub2txt.c                  
src/epub2txt.c:492:15: error: call to undeclared function 'canonicalize_file_name'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  492 |         opf = canonicalize_file_name (tmp);
      |               ^                              
src/epub2txt.c:492:13: error: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Wint-conversion]
  492 |         opf = canonicalize_file_name (tmp);
      |             ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~   
src/epub2txt.c:536:12: error: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Wint-conversion]
  536 |               opf = canonicalize_file_name (tmp);                                                   
      |                   ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 errors generated.
make: *** [Makefile:24: build/epub2txt.o] Error 1
~/tools/epub2txt2 $
```
</details>

Querela commented:
Maybe the comment should also be changed in [`src/util.c`](https://github.com/kevinboone/epub2txt2/blob/69a2c2f78c0359d780b07cce353869c27e7a4328/src/util.c#L88) to avoid confusion ...

kevinboone commented:
This is a tricky situation. The `canonicalize_file_path` section was added at the request of another user, to prevent a badly-formatted EPUB being used to read the filesystem. I was reluctant to merge this change, because I had a feeling that it would lead to problems on other platforms -- and it seems that it has.

Do you think that your proposed change will also address the earlier problem? I can check myself, but I don't really have time at present.

Querela commented:
Hm. Well, it works for me locally with the patch, so I'm happy, just wanted to contribute back.
I'm not sure if the `canonicalize_file_path` method is just a simple wrapper around `realpath(path, NULL)` [1] or if more is going on. And I'm really not sure how different systems implement this and why it's not there on Termux. The man pages are clear but again, do they differ between systems?!
So, better keep it as is if it's just a really tiny edge case with Termux.
We could add some comments or plattform dependant macros but that's overkill...

[1] https://codebrowser.dev/glibc/glibc/stdlib/canonicalize.c.html#__canonicalize_file_name

Querela commented:
Pinging @mebeim 🙂
Are you aware of unknown side effects of changing `canonicalize_file_path(path)` to `realpath(path, NULL)` which should be equivalent calls?

mebeim commented:
Wow first time I see someone building stuff in Termux, pretty cool. You bring up very valid points and in fact `canonicalize_file_name()` was a poor choice on my part. Indeed it seems like `canonicalize_file_name()` is practically equivalent to `realpath()`, but the former is GNU and the latter is POSIX. 

@Querela I think your patch makes sense (you have a typo in the commit msg s/path/name/). I think you may also want to add `#define _XOPEN_SOURCE 500` at the top (I see it mentioned [in the man page](https://manned.org/man/realpath.3) at least for glibc).

Querela commented:
Ok. Added this macro definition on top of `src/epub2txt.c`. Also updated the comment in `src/util.c` that refers to the old method.
Amended the original commit.

mebeim commented:
I give it [6 to 8 weeks](https://meta.stackexchange.com/a/19514/288337).
